builder/frameworks/espidf.py

```
@@ -419,7 +419,11 @@ def generate_board_specific_config():
         # Fallback to board.json manifest
         if not flash_size:
             flash_size = board.get("upload", {}).get("flash_size", None)
-        
+
+        if flash_size == "2MB":
+            print(f"Info: Detected 2MB flash size setting, override to 4MB for Arduino MMU page size compatibility")
```

Ruff (0.15.1)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)